src/main/java/ti4/generator/UnitRenderGenerator.java

```diff
@@ -0,0 +1,545 @@
+        if (unitKey.getUnitType() == UnitType.Warsun && ThreadLocalRandom.current().nextInt(1000) == 0) {
+            String spoopypath = resourceHelper.getSpoopyFile();
+            spoopy = ImageHelper.read(spoopypath);
+        }
+
+        // Ghemina special units
+        if (unitKey.getUnitType() == UnitType.Lady) {
+            String name = "units_ds_ghemina_lady_wht.png";
+            String spoopyPath = resourceHelper.getDecalFile(name);
+            spoopy = ImageHelper.read(spoopyPath);
+        }
+        if (unitKey.getUnitType() == UnitType.Flagship && player.ownsUnit("ghemina_flagship_lord")) {
+            String name = "units_ds_ghemina_lord_wht.png";
+            String spoopyPath = resourceHelper.getDecalFile(name);
+            spoopy = ImageHelper.read(spoopyPath);
+        }
+
+        return spoopy;
+    }
+
+    private Integer getBulkUnitCount(UnitKey unitKey, int unitCount) {
+        return Set.of(UnitType.Fighter, UnitType.Infantry).contains(unitKey.getUnitType()) ? unitCount : null;
+    }
+
+    private Map<UnitKey, Integer> sortUnits(Map<UnitKey, Integer> tempUnits) {
+        Map<UnitKey, Integer> sortedUnits = new LinkedHashMap<>();
+
+        // Add mechs first
+        for (Map.Entry<UnitKey, Integer> entry : tempUnits.entrySet()) {
+            UnitKey id = entry.getKey();
+            if (id != null && id.getUnitType() == UnitType.Mech) {
+                sortedUnits.put(id, entry.getValue());
+            }
+        }
+
+        // Remove mechs from temp units
+        for (UnitKey key : sortedUnits.keySet()) {
+            tempUnits.remove(key);
+        }
+
+        // Add remaining units
+        sortedUnits.putAll(tempUnits);
+        return sortedUnits;
+    }
+
+    private Point calculateUnitPosition(
+        PositioningContext posCtx,
+        UnitKey unitKey,
+        int count) {
+        Point predefinedPosition = getTokenPosition(unitKey, posCtx);
+
+        // Handle space units (non-fighter/infantry units in space with predefined positions)
+        if (ctx.isSpace && predefinedPosition != null && !posCtx.fighterOrInfantry) {
+            predefinedPosition.translate(ctx.spaceUnitXOffset * count, ctx.spaceUnitYOffset * count);
+            return predefinedPosition;
+        }
+
+        // Use predefined token position if available
+        if (predefinedPosition != null) {
+            return predefinedPosition;
+        }
+
+        // Default to spiral positioning
+        return calculateSpiralPosition(posCtx, radius, degree, degreeChange, rectangles);
+    }
+
+    private Point getTokenPosition(UnitKey unitKey, PositioningContext posCtx) {
+        boolean fighterOrInfantry = Set.of(UnitType.Infantry, UnitType.Fighter).contains(unitKey.getUnitType());
+        String positionKey = fighterOrInfantry ? "tkn_" + posCtx.unitId : posCtx.unitId;
+        return ctx.unitTokenPosition.getPosition(positionKey);
+    }
+
+    private Point calculateSpiralPosition(
```

Comment:
I'll attempt a draw of all prod games and check to see if we enter this method at all